kytos/core/interface.py

```diff
@@ -195,148 +225,58 @@ def status_reason(self):
-                update = True
-            result.append([start_a, end_a])
-            a_i += 1
-
-        return result
-
     def set_tag_ranges(self, tag_ranges: list[list[int]], tag_type: str):
-        """Set new restriction"""
+        """Set new restriction, tag_ranges."""
         if tag_type != TAGType.VLAN.value:
             msg = f"Tag type {tag_type} is not supported."
             raise KytosTagtypeNotSupported(msg)
         with self._tag_lock:
-            used_tags = self.range_difference(
+            used_tags = range_difference(
                 self.tag_ranges[tag_type], self.available_tags[tag_type]
             )
             # Verify new tag_ranges
-            missing = self.range_difference(used_tags, tag_ranges)
+            missing = range_difference(used_tags, tag_ranges)
             if missing:
                 msg = f"Missing tags in tag_range: {missing}"
                 raise KytosSetTagRangeError(msg)
 
             # Resizing
-            new_tag_ranges = self.range_difference(
+            new_available_tags = range_difference(
                 tag_ranges, used_tags
             )
-            self.available_tags[tag_type] = new_tag_ranges
+            self.available_tags[tag_type] = new_available_tags
             self.tag_ranges[tag_type] = tag_ranges
 
     def remove_tag_ranges(self, tag_type: str):
-        """Set tag_ranges[tag_type] to default [[1, 4095]]"""
+        """Set tag_ranges[tag_type] to default value"""
         if tag_type != TAGType.VLAN.value:
             msg = f"Tag type {tag_type} is not supported."
             raise KytosTagtypeNotSupported(msg)
         with self._tag_lock:
-            used_tags = self.range_difference(
+            used_tags = range_difference(
                 self.tag_ranges[tag_type], self.available_tags[tag_type]
             )
-            self.available_tags[tag_type] = self.range_difference(
+            self.available_tags[tag_type] = range_difference(
                 self.default_tag_values[tag_type], used_tags
             )
             self.tag_ranges[tag_type] = self.default_tag_values[tag_type]
 
-    @staticmethod
-    def find_index_remove(
-        available_tags: list[list[int]],
-        tag_range: list[int]
-    ) -> Optional[int]:
-        """Find the index of tags in available_tags to be removed"""
-        index = bisect.bisect_left(available_tags, tag_range)
-        if (index < len(available_tags) and
-                tag_range[0] >= available_tags[index][0] and
-                tag_range[1] <= available_tags[index][1]):
-            return index
-
-        if (index-1 > -1 and
-                tag_range[0] >= available_tags[index-1][0] and
-                tag_range[1] <= available_tags[index-1][1]):
-            return index - 1
-
-        return None
-
     def remove_tags(self, tags: list[int], tag_type: str = 'vlan') -> bool:
-        """Remove tags by resize available_tags
+        """Remove tags by resizing available_tags
```

Comment:
`remove_tags` is a lower-ish level method resizing or updating `available_tags`, should we rename this as `_remove_tags` to signal that it shouldn't be called directly via NApps or other packages/modules, and mostly via `use_tags`? Same rationale as you've done for `_use_tags` that's being called on `use_tags`. That way making it clear for whoever is looking to use a tag not to use `remove_tags` directly, unless they know what they're doing.

Reply:
Changed to `_remove_tags()` and, also, changed `add_tags()` to `_add_tags()`.

Reply:
Looks good. Great.